src/client/ManagedNetwork.js

```diff
@@ -206,17 +206,26 @@ export default class ManagedNetwork {
         this._removeDeadNodes();
         this._readmitNodes();
 
-        /** @type {NetworkNodeT[]} */
         const nodes = [];
-        let healthyNodes = this._healthyNodes;
+        // Create a shallow for safe iteration
+        let healthyNodes = this._healthyNodes.slice();
+        count = Math.min(count, healthyNodes.length);
+
         for (let i = 0; i < count; i++) {
+            // Select a random index
             const nodeIndex = Math.floor(Math.random() * healthyNodes.length);
             const selectedNode = healthyNodes[nodeIndex];
+
+            // Check if the node exists
+            if (!selectedNode) {
+                console.error("Selected an undefined node. Index:", nodeIndex);
```

Comment:
Do we need to log in here? Let's put a logger with an appropriate level of debugging.